dispatch/dispatch.go

```diff
@@ -271,7 +271,9 @@ func (d *Dispatcher) processAlert(alert *types.Alert, route *Route) {
 	ag, ok := group[fp]
```

Comment:
This part should probably be protected too. It might be simpler to just add a `defer` after the first lock (L262):

```go
fp := groupLabels.Fingerprint()
 
d.mtx.Lock()
defer d.mtx.Unlock()
```